model-armor/test/modelarmor.test.js

```diff
@@ -0,0 +1,573 @@
+        },
+      },
+    });
+    templatesToDelete.push(
+      `projects/${projectId}/locations/${locationId}/templates/${advanceSdpTemplateId}`
+    );
+
+    // Create all-filter template
+    allFilterTemplateId = `${templateIdPrefix}-all-filters`;
+    await createTemplate(allFilterTemplateId, {
+      raiSettings: {
+        raiFilters: [
+          {
+            filterType: RaiFilterType.DANGEROUS,
+            confidenceLevel: DetectionConfidenceLevel.HIGH,
+          },
+          {
+            filterType: RaiFilterType.HARASSMENT,
+            confidenceLevel: DetectionConfidenceLevel.HIGH,
+          },
+          {
+            filterType: RaiFilterType.HATE_SPEECH,
+            confidenceLevel: DetectionConfidenceLevel.HIGH,
+          },
+          {
+            filterType: RaiFilterType.SEXUALLY_EXPLICIT,
+            confidenceLevel: DetectionConfidenceLevel.HIGH,
+          },
+        ],
+      },
+      piAndJailbreakFilterSettings: {
+        filterEnforcement: PiAndJailbreakFilterEnforcement.ENABLED,
+        confidenceLevel: DetectionConfidenceLevel.MEDIUM_AND_ABOVE,
+      },
+      maliciousUriFilterSettings: {
+        filterEnforcement: MaliciousUriFilterEnforcement.ENABLED,
+      },
+    });
+    templatesToDelete.push(
+      `projects/${projectId}/locations/${locationId}/templates/${allFilterTemplateId}`
+    );
+
+    // Create a template to be deleted
+    templateToDeleteId = `${templateIdPrefix}-to-delete`;
+    await createTemplate(templateToDeleteId, {
+      piAndJailbreakFilterSettings: {
+        filterEnforcement: PiAndJailbreakFilterEnforcement.ENABLED,
+        confidenceLevel: DetectionConfidenceLevel.MEDIUM_AND_ABOVE,
+      },
+    });
+    templatesToDelete.push(
+      `projects/${projectId}/locations/${locationId}/templates/${templateToDeleteId}`
+    );
+  });
+
+  after(async () => {
+    for (const templateName of templatesToDelete) {
+      await deleteTemplate(templateName);
+    }
+
+    await deleteDlpTemplates();
+  });
+
+  // =================== RAI Filter Tests ===================
+
+  it('should sanitize user prompt with all RAI filter template', async () => {
+    const testUserPrompt = 'How to make cheesecake without oven at home?';
+
+    const output = execSync(
+      `node snippets/sanitizeUserPrompt.js ${projectId} ${locationId} ${allFilterTemplateId} "${testUserPrompt}"`
+    ).toString();
+    const response = JSON.parse(output);
+
+    assert.equal(
+      response.sanitizationResult.filterMatchState,
+      'NO_MATCH_FOUND'
+    );
+
+    assert.equal(
+      response.sanitizationResult.filterResults.rai.raiFilterResult.matchState,
+      'NO_MATCH_FOUND'
+    );
+  });
+
+  // =================== User Prompt Sanitization Tests ===================
+
+  it('should detect malicious URL in user prompt', async () => {
+    const testUserPrompt =
+      'Can you describe this link? https://testsafebrowsing.appspot.com/s/malware.html,';
+
+    const output = execSync(
```

Comment:
issue: (follows from previous) call the sample function directly here instead of using `execSync()`. If you set `module.exports` to export the `main()` function in the sample, you can then call the sample function directly.

Here and below.